src/Tests/Microsoft.DotNet.TemplateLocator.Tests/GivenAnMSBuildSdkResolver.cs

```diff
@@ -0,0 +1,35 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using FluentAssertions;
+using System.IO;
+using Xunit;
+using Xunit.Abstractions;
+using Microsoft.NET.TestFramework;
+using System.Linq;
+using NuGet.Common;
+using NuGet.Protocol;
+
+namespace Microsoft.DotNet.TemplateLocator.Tests
+{
+    public class GivenAnTemplateLocator : SdkTest
+    {
+
+        public GivenAnTemplateLocator(ITestOutputHelper logger) : base(logger)
+        {
+        }
+
+        [Fact]
+        public void ItShouldReturnListOfTemplates()
+        {
+            var resolver = new TemplateLocator();
+            var stage2Dotnet = new DirectoryInfo(TestContext.Current.ToolsetUnderTest.DotNetHostPath);
+            var stage2Templates = stage2Dotnet.Parent.GetDirectories("templates")[0].EnumerateDirectories().First();
```

Comment:
If the directory doesn't exist, won't GetDirectories() return an empty array and then trigger an exception when you look at the [0] item

Reply:
That's fine. The only user see it is the dev working on _dotnet/templating_. And they can react.